src/mono/mono/mini/mini-runtime.c

```diff
@@ -4383,6 +4383,30 @@ mini_llvm_init (void)
 #endif
 }
 
+#ifdef ENSURE_PRIMARY_STACK_SIZE
+/*++ 
+ Function: 
+   EnsureStackSize 
+  
+ Abstract: 
+   This fixes a problem on MUSL where the initial stack size reported by the 
+   pthread_attr_getstack is about 128kB, but this limit is not fixed and 
```

Comment:
When does the stack size get reported as 128kB?

If I run the following code in an Alpine container, it reports the size to be 8MB (8388608), as set when the thread was created.

```
#include <pthread.h>
#include <stdio.h>

void* thread_fn(void*)
{
    printf("hello from thread\n");
    return 0;
}

int main()
{
    pthread_attr_t attr;
    pthread_t thread;
    int res;
    size_t set_stack_size = 8*1024*1024;
    size_t stack_size;

    res = pthread_attr_init (&attr);

    res = pthread_attr_setstacksize (&attr, set_stack_size);

    res = pthread_create (&thread, &attr, thread_fn, NULL);

    res = pthread_attr_getstacksize (&attr, &stack_size);

    pthread_join(thread, NULL);

    printf("%zu\n", stack_size);

    return 0;
}

```

Reply:
nvmd, here it is:
```c
#define _GNU_SOURCE   
#include <pthread.h>
#include <stdio.h>

int main()
{
    pthread_attr_t attr;
    pthread_t thread = pthread_self();

    res = pthread_attr_init (&attr);

    res = pthread_getattr_np (thread, &attr);

    res = pthread_attr_getstacksize (&attr, &stack_size);

    printf("%zu\n", stack_size);

    return 0;
}
```

```
$ ./a.out 
131072
```